src/multi_select.coffee

```diff
@@ -98,12 +115,30 @@ Ember.Widgets.MultiSelectComponent = Ember.Widgets.SelectComponent.extend
     @set 'values', Ember.A [] unless @get('values')
 
   deletePressed: (event) ->
-    if event.target.selectionStart == 0
+    # check if the cursor is at the beginning and no text is selected
+    if event.target.selectionStart == 0 and event.target.selectionEnd == 0
       @removeSelectItem(@get('selections.lastObject'))
 
   removeSelectItem: (item) ->
     @get('selections').removeObject item
 
+  focusIn: (event) ->
+    @addBorderMultiSelectContainer()
+
+  focusOut: (event) ->
+    @removeBorderMultiSelectContainer()
+
+  keyDown: (event) ->
+    @_super(event)
+    activeElement = $(document.activeElement)[0]
+
+    # dynamically add css style to highlight the border if any elements inside
+    # the multiselect is focused
+    if @$()[0].contains(activeElement) or @$()[0]==activeElement
```

Comment:
is this code necessary? I tried deleting it and the border still seemed to appear/disappear as expected. If it's necessary please add a comment explaining the case that breaks without it